File: ios/security_toolkit/Sources/security_toolkit/SecurityToolkitPlugin.swift
import Flutter
import UIKit

public class SecurityToolkitPlugin: NSObject, FlutterPlugin {
    
    let securityToolkit: SecurityToolkitApi
    
    override init() {
        self.securityToolkit = ServiceLocator.getSecurityToolkitApi()
    }
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let plugin = SecurityToolkitPlugin()
        let securityToolkit = plugin.securityToolkit
        SecurityToolkitApiSetup.setUp(binaryMessenger: registrar.messenger(), api: securityToolkit)
        registrar.publish(plugin)
    }
}

File: android/src/main/kotlin/com/apadmi/security_toolkit/SecurityToolkitPlugin.kt
package com.apadmi.security_toolkit

import SecurityToolkitApi
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class SecurityToolkitPlugin: FlutterPlugin {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    val securityApi = ServiceLocator.getSecurityToolkit(flutterPluginBinding.applicationContext)
    SecurityToolkitApi.setUp(flutterPluginBinding.binaryMessenger, securityApi)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    // Intentionally blank.
  }
}
